File: Weather/Weather/Data Models/ForecastFullDay.swift
//
//  FullDayForcast.swift
//  Weather
//
//  Created by Pasan Diksura on 2024-11-23.
//

import Foundation

struct ForecastFullDayDTO: Decodable, Hashable, Identifiable {
    var id: String = UUID().uuidString
    var date: String = ""
    let maxtempC: Double
    let maxtempF: Double
    let mintempC: Double
    let mintempF: Double
    let avgtempC: Double
    let avgtempF: Double
    let maxwindMph: Double
    let maxwindKph: Double
    let totalprecipMm: Double
    let totalprecipIn: Double
    let totalsnowCm: Double
    let avgvisKm: Double
    let avgvisMiles: Double
    let avghumidity: Double
    let dailyWillItRain: Int
    let dailyChanceOfRain: Double
    let dailyWillItSnow: Int
    let dailyChanceOfSnow: Double
    let condition: ConditionDTO
    let uv: Double
    let airQuality: AirQualityDTO
    
    enum CodingKeys: String, CodingKey {
        case maxtempC = "maxtemp_c"
        case maxtempF = "maxtemp_f"
        case mintempC = "mintemp_c"
        case mintempF = "mintemp_f"
        case avgtempC = "avgtemp_c"
        case avgtempF = "avgtemp_f"
        case maxwindMph = "maxwind_mph"
        case maxwindKph = "maxwind_kph"
        case totalprecipMm = "totalprecip_mm"
        case totalprecipIn = "totalprecip_in"
        case totalsnowCm = "totalsnow_cm"
        case avgvisKm = "avgvis_km"
        case avgvisMiles = "avgvis_miles"
        case avghumidity = "avghumidity"
        case dailyWillItRain = "daily_will_it_rain"
        case dailyChanceOfRain = "daily_chance_of_rain"
        case dailyWillItSnow = "daily_will_it_snow"
        case dailyChanceOfSnow = "daily_chance_of_snow"
        case condition
        case uv
        case airQuality = "air_quality"
    }
}

File: WeatherServer/WeatherServer.API/Models/ForecastFullDay.cs
using System.Text.Json.Serialization;
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace WeatherServer.API.Models
{
    public record ForecastFullDay
        (
            //[property: JsonPropertyName("hour")] string date,
            [property: JsonPropertyName("maxtemp_c")] double maxtempC,
            [property: JsonPropertyName("maxtemp_f")] double maxtempF,
            [property: JsonPropertyName("mintemp_c")] double mintempC,
            [property: JsonPropertyName("mintemp_f")] double mintempF,
            [property: JsonPropertyName("avgtemp_c")] double avgtempC,
            [property: JsonPropertyName("avgtemp_f")] double avgtempF,
            [property: JsonPropertyName("maxwind_mph")] double maxwindMph,
            [property: JsonPropertyName("maxwind_kph")] double maxwindKph,
            [property: JsonPropertyName("totalprecip_mm")] double totalprecipMm,
            [property: JsonPropertyName("totalprecip_in")] double totalprecipIn,
            [property: JsonPropertyName("totalsnow_cm")] double totalsnowCm,
            [property: JsonPropertyName("avgvis_km")] double avgvisKm,
            [property: JsonPropertyName("avgvis_miles")] double avgvisMiles,
            [property: JsonPropertyName("avghumidity")] double avghumidity,
            [property: JsonPropertyName("daily_will_it_rain")] int dailyWillItRain,
            [property: JsonPropertyName("daily_chance_of_rain")] double dailyChanceOfRain,
            [property: JsonPropertyName("daily_will_it_snow")] int dailyWillItSnow,
            [property: JsonPropertyName("daily_chance_of_snow")] double dailyChanceOfSnow,
            Condition condition,
            double uv,
            [property: JsonPropertyName("air_quality")] AirQuality airQuality
        );
}
